tests/bpf2c_tests/elf_bpf.cpp

```diff
@@ -101,37 +127,36 @@ run_test_elf(const std::string& elf_file, bool verify, bool expect_failure = fal
     test("--sys", "_sys.txt");
 }
 
-#define DECLARE_TEST(FILE, VERIFY) \
-    TEST_CASE(FILE, "[elf_bpf_code_gen]") { run_test_elf(FILE ".o", VERIFY); }
-
-#define DECLARE_TEST_VERIFICATION_FAILURE(FILE) \
-    TEST_CASE(FILE " verification failure", "[elf_bpf_code_gen]") { run_test_elf(FILE ".o", true, true); }
-
-DECLARE_TEST("bindmonitor", true)
-DECLARE_TEST("bindmonitor_ringbuf", true)
-DECLARE_TEST("bindmonitor_tailcall", true)
-DECLARE_TEST("bpf", true)
-DECLARE_TEST("bpf_call", true)
-DECLARE_TEST("cgroup_sock_addr", true)
-DECLARE_TEST("decap_permit_packet", true)
-DECLARE_TEST("divide_by_zero", true)
-DECLARE_TEST("droppacket", true)
-DECLARE_TEST("encap_reflect_packet", true)
-DECLARE_TEST("map", false)
-DECLARE_TEST("map_in_map", true)
-DECLARE_TEST("map_in_map_v2", true)
-DECLARE_TEST("map_reuse", true)
-DECLARE_TEST("map_reuse_2", true)
-DECLARE_TEST("printk", true)
-DECLARE_TEST("printk_legacy", true)
-DECLARE_TEST("reflect_packet", true)
-DECLARE_TEST("tail_call", true)
-DECLARE_TEST("tail_call_bad", true)
-DECLARE_TEST("tail_call_map", true)
-DECLARE_TEST("tail_call_multiple", true)
-DECLARE_TEST("test_sample_ebpf", true)
-DECLARE_TEST("test_utility_helpers", true)
-DECLARE_TEST("sockops", true)
-
-DECLARE_TEST_VERIFICATION_FAILURE("droppacket_unsafe")
-DECLARE_TEST_VERIFICATION_FAILURE("printk_unsafe")
+#define DECLARE_TEST(FILE, MODE) \
+    TEST_CASE(FILE " " #MODE, "[elf_bpf_code_gen]") { run_test_elf(FILE ".o", MODE); }
+
+DECLARE_TEST("bindmonitor", _test_mode::Verify)
+DECLARE_TEST("bindmonitor_ringbuf", _test_mode::Verify)
+DECLARE_TEST("bindmonitor_tailcall", _test_mode::Verify)
+DECLARE_TEST("bpf", _test_mode::Verify)
+DECLARE_TEST("bpf_call", _test_mode::Verify)
+DECLARE_TEST("cgroup_sock_addr", _test_mode::Verify)
+DECLARE_TEST("decap_permit_packet", _test_mode::Verify)
+DECLARE_TEST("divide_by_zero", _test_mode::Verify)
+DECLARE_TEST("droppacket", _test_mode::Verify)
+DECLARE_TEST("encap_reflect_packet", _test_mode::Verify)
+DECLARE_TEST("map", _test_mode::NoVerify)
+DECLARE_TEST("map_in_map", _test_mode::Verify)
+DECLARE_TEST("map_in_map_v2", _test_mode::Verify)
+DECLARE_TEST("map_reuse", _test_mode::Verify)
+DECLARE_TEST("map_reuse_2", _test_mode::Verify)
+DECLARE_TEST("printk", _test_mode::Verify)
+DECLARE_TEST("printk_legacy", _test_mode::Verify)
+DECLARE_TEST("reflect_packet", _test_mode::Verify)
+DECLARE_TEST("tail_call", _test_mode::Verify)
+DECLARE_TEST("tail_call_bad", _test_mode::Verify)
+DECLARE_TEST("tail_call_map", _test_mode::Verify)
+DECLARE_TEST("tail_call_multiple", _test_mode::Verify)
+DECLARE_TEST("test_sample_ebpf", _test_mode::Verify)
+DECLARE_TEST("test_utility_helpers", _test_mode::Verify)
+DECLARE_TEST("sockops", _test_mode::Verify)
+
+DECLARE_TEST("droppacket_unsafe", _test_mode::VerifyFail)
+DECLARE_TEST("printk_unsafe", _test_mode::VerifyFail)
+DECLARE_TEST("no_such_file", _test_mode::FileNotFound)
+DECLARE_TEST("bpf", _test_mode::UseHash)
```

Comment:
"bpf" is being tested with mode "UseHash", but the generated expected files for "bpf" contain the following (hash is NULL):

`static void _get_hash(_Outptr_result_buffer_maybenull_(*size) const uint8_t** hash, _Out_ size_t* size)
{
	*hash = NULL;
	*size = 0;
}`

Is this expected?

Reply:
Yes, for the "UseHash" case it just verifies that something is generated and the main function returns success (the hash will be different depending on the path it's built from).